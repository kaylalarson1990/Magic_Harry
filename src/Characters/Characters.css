.characterCard {
    align-items: center;
    background-color: white;
    border: .5px solid #EBECEC;
    border-radius: 3px;
    box-shadow: 2px 2px lightgrey;
    display: flex;
    flex-direction: column;
    font-family: 'Open Sans';
    height: auto;
    margin: 15px;
    width: 300px;
}

h2 {
    text-align: center;
    font-weight: 600;
}

.favorite-btn {
    margin-top: 10px;
    width: 35px;
}