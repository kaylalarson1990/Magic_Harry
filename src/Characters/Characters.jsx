import React, { Component } from "react";
import PropTypes from "prop-types";
import ReactCardFlip from "react-card-flip";
import inactive from "../images/inactive.png";
import active from "../images/active.png";
import "./Characters.css";

export class Characters extends Component {
  constructor() {
    super();
    this.state = {
      favorited: false,
      isFlipped: false,
      error: ""
    };
  }

  addFavorite = () => {
    this.state.favorited
      ? this.setState({ favorited: false })
      : this.setState({ favorited: true });
  };

  handleClick = e => {
    e.preventDefault();
    this.setState(prevState => ({ isFlipped: !prevState.isFlipped }));
  };

  render() {
    const { name, role, house, school, id, species } = this.props;
    return (
      <ReactCardFlip isFlipped={this.state.isFlipped} flipDirection="vertical">
        <article className="characterCard" key="front">
          <h2>{name}</h2>
          <button onClick={this.handleClick}>
            Click here for more details
          </button>
          <img
            src={this.state.favorited ? active : inactive}
            alt="favorite icon"
            className="favorite-btn"
            onClick={this.addFavorite}
          />
        </article>
        <article className="characterCard" key="back">
          <p><span>Role:</span> {role}</p>
          <p><span>House:</span> {house}</p>
          <p><span>School:</span> {school}</p>
          <p><span>Species:</span> {species}</p>
          <button onClick={this.handleClick}>Return to Front</button>
        </article>
      </ReactCardFlip>
    );
  }
}

Characters.propTypes = {
  name: PropTypes.string,
  role: PropTypes.string,
  house: PropTypes.string,
  species: PropTypes.string,
  id: PropTypes.number
};

export default Characters;
