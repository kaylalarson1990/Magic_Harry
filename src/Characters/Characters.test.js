import React from "react";
import { shallow } from "enzyme";
import Characters from "./Characters";
//... 23,24,25,29,30

describe("Characters", () => {
  let wrapper, mockFunc, instance;
  let event = {
    preventDefault: jest.fn()
  };

  let props = {
    name: "Kayla",
    role: "Student",
    house: "Apartment",
    school: "Turing",
    id: 1,
    species: "Human"
  };

  beforeEach(() => {
    mockFunc = jest.fn();
    wrapper = shallow(<Characters {...props} />);
    instance = wrapper.instance();
  });

  it("should match snapshot", () => {
    wrapper.setState({ error: "" });
    expect(wrapper).toMatchSnapshot();
  });

  it("should be able to toggle favorite", () => {
    instance.addFavorite = mockFunc;
    wrapper.find(".favorite-btn").simulate("click");
    expect(instance.addFavorite).not.toHaveBeenCalled();
  });

  it("should set flipped to state", () => {
    expect(wrapper.state("isFlipped")).toEqual(false);
    instance.handleClick(event);
    expect(wrapper.state("isFlipped")).toEqual(true);
  });

  it("should set favorite to state", () => {
    expect(wrapper.state("favorited")).toEqual(false);
    instance.addFavorite();
    expect(wrapper.state("favorited")).toEqual(true);
  });
});
