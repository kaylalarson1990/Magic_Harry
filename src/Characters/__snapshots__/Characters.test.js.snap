// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters should match snapshot 1`] = `
<t
  cardStyles={
    Object {
      "back": Object {},
      "front": Object {},
    }
  }
  cardZIndex="auto"
  containerStyle={Object {}}
  flipDirection="vertical"
  flipSpeedBackToFront={0.6}
  flipSpeedFrontToBack={0.6}
  infinite={false}
  isFlipped={false}
>
  <article
    className="characterCard"
    key="front"
  >
    <h2>
      Kayla
    </h2>
    <button
      onClick={[Function]}
    >
      Click here for more details
    </button>
    <img
      alt="favorite icon"
      className="favorite-btn"
      onClick={[Function]}
      src="inactive.png"
    />
  </article>
  <article
    className="characterCard"
    key="back"
  >
    <p>
      <span>
        Role:
      </span>
       
      Student
    </p>
    <p>
      <span>
        House:
      </span>
       
      Apartment
    </p>
    <p>
      <span>
        School:
      </span>
       
      Turing
    </p>
    <p>
      <span>
        Species:
      </span>
       
      Human
    </p>
    <button
      onClick={[Function]}
    >
      Return to Front
    </button>
  </article>
</t>
`;
